### DISCUSSION TITLE
Integrate @joint/core in reactjs

### DISCUSSION THREAD

**Discussion by rajeshK88** [clientIO/joint#2540] [Q&A]
### Introduction

What are the steps to integrate @joint/core with reactjs. 
Reactjs state should not impact the existing diagram. 

### Does your question relate to JointJS or JointJS+. Select both if applicable.

JointJS

**Comment by kumilingus** [comment]
You can check our `JointJS+` tutorial: https://resources.jointjs.com/tutorial/react-ts

Here's a simple [example](https://codesandbox.io/p/devbox/baalakay-node-react-forked-2hhcy5?file=%2Fsrc%2FApp.js%3A22%2C34&layout=%257B%2522sidebarPanel%2522%253A%2522EXPLORER%2522%252C%2522rootPanelGroup%2522%253A%257B%2522direction%2522%253A%2522horizontal%2522%252C%2522contentType%2522%253A%2522UNKNOWN%2522%252C%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522id%2522%253A%2522ROOT_LAYOUT%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522UNKNOWN%2522%252C%2522direction%2522%253A%2522vertical%2522%252C%2522id%2522%253A%2522clsqxqd6i00083b6itdcrmxdz%2522%252C%2522sizes%2522%253A%255B70%252C30%255D%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522EDITOR%2522%252C%2522direction%2522%253A%2522horizontal%2522%252C%2522id%2522%253A%2522EDITOR%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522EDITOR%2522%252C%2522id%2522%253A%2522clsqxqd6i00033b6iil8qiu5c%2522%257D%255D%257D%252C%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522SHELLS%2522%252C%2522direction%2522%253A%2522horizontal%2522%252C%2522id%2522%253A%2522SHELLS%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522SHELLS%2522%252C%2522id%2522%253A%2522clsqxqd6i00053b6ivknllgky%2522%257D%255D%252C%2522sizes%2522%253A%255B100%255D%257D%255D%257D%252C%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522DEVTOOLS%2522%252C%2522direction%2522%253A%2522vertical%2522%252C%2522id%2522%253A%2522DEVTOOLS%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522DEVTOOLS%2522%252C%2522id%2522%253A%2522clsqxqd6i00073b6ie9mwiyfc%2522%257D%255D%252C%2522sizes%2522%253A%255B100%255D%257D%255D%252C%2522sizes%2522%253A%255B50%252C50%255D%257D%252C%2522tabbedPanels%2522%253A%257B%2522clsqxqd6i00033b6iil8qiu5c%2522%253A%257B%2522tabs%2522%253A%255B%257B%2522id%2522%253A%2522clsqxqd6i00023b6ium36wr5i%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522type%2522%253A%2522FILE%2522%252C%2522initialSelections%2522%253A%255B%257B%2522startLineNumber%2522%253A22%252C%2522startColumn%2522%253A34%252C%2522endLineNumber%2522%253A22%252C%2522endColumn%2522%253A34%257D%255D%252C%2522filepath%2522%253A%2522%252Fsrc%252FApp.js%2522%252C%2522state%2522%253A%2522IDLE%2522%257D%255D%252C%2522id%2522%253A%2522clsqxqd6i00033b6iil8qiu5c%2522%252C%2522activeTabId%2522%253A%2522clsqxqd6i00023b6ium36wr5i%2522%257D%252C%2522clsqxqd6i00073b6ie9mwiyfc%2522%253A%257B%2522tabs%2522%253A%255B%257B%2522id%2522%253A%2522clsqxqd6i00063b6i7w6hursl%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522type%2522%253A%2522TASK_PORT%2522%252C%2522taskId%2522%253A%2522start%2522%252C%2522port%2522%253A3000%252C%2522path%2522%253A%2522%252F%2522%257D%255D%252C%2522id%2522%253A%2522clsqxqd6i00073b6ie9mwiyfc%2522%252C%2522activeTabId%2522%253A%2522clsqxqd6i00063b6i7w6hursl%2522%257D%252C%2522clsqxqd6i00053b6ivknllgky%2522%253A%257B%2522tabs%2522%253A%255B%257B%2522id%2522%253A%2522clsqxqd6i00043b6ijrxc569y%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522type%2522%253A%2522TASK_LOG%2522%252C%2522taskId%2522%253A%2522start%2522%257D%255D%252C%2522id%2522%253A%2522clsqxqd6i00053b6ivknllgky%2522%252C%2522activeTabId%2522%253A%2522clsqxqd6i00043b6ijrxc569y%2522%257D%257D%252C%2522showDevtools%2522%253Atrue%252C%2522showShells%2522%253Atrue%252C%2522showSidebar%2522%253Atrue%252C%2522sidebarPanelSize%2522%253A15%257D) using only `@joint/core`.

**Reply by rajeshK88** [reply]
Got it.. thanks for clarifying. 
1. is there any support for reactjs in future like separate component ?
2. Looks some of the jointjs core components are not working while react state is updated.
3. also double mesh is not working in @joint/core.

**Reply by kumilingus** [reply]
> is there any support for reactjs in future like separate component ?

Yes, currently being developed (similar to this https://github.com/clientIO/joint/pull/2522).

> also double mesh is not working in @joint/core.

It works. See [example](https://codesandbox.io/p/devbox/baalakay-node-react-forked-2hhcy5?file=%2Fsrc%2FApp.js%3A22%2C34&layout=%257B%2522sidebarPanel%2522%253A%2522EXPLORER%2522%252C%2522rootPanelGroup%2522%253A%257B%2522direction%2522%253A%2522horizontal%2522%252C%2522contentType%2522%253A%2522UNKNOWN%2522%252C%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522id%2522%253A%2522ROOT_LAYOUT%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522UNKNOWN%2522%252C%2522direction%2522%253A%2522vertical%2522%252C%2522id%2522%253A%2522clsqxqd6i00083b6itdcrmxdz%2522%252C%2522sizes%2522%253A%255B70%252C30%255D%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522EDITOR%2522%252C%2522direction%2522%253A%2522horizontal%2522%252C%2522id%2522%253A%2522EDITOR%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522EDITOR%2522%252C%2522id%2522%253A%2522clsqxqd6i00033b6iil8qiu5c%2522%257D%255D%257D%252C%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522SHELLS%2522%252C%2522direction%2522%253A%2522horizontal%2522%252C%2522id%2522%253A%2522SHELLS%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522SHELLS%2522%252C%2522id%2522%253A%2522clsqxqd6i00053b6ivknllgky%2522%257D%255D%252C%2522sizes%2522%253A%255B100%255D%257D%255D%257D%252C%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522DEVTOOLS%2522%252C%2522direction%2522%253A%2522vertical%2522%252C%2522id%2522%253A%2522DEVTOOLS%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522DEVTOOLS%2522%252C%2522id%2522%253A%2522clsqxqd6i00073b6ie9mwiyfc%2522%257D%255D%252C%2522sizes%2522%253A%255B100%255D%257D%255D%252C%2522sizes%2522%253A%255B50%252C50%255D%257D%252C%2522tabbedPanels%2522%253A%257B%2522clsqxqd6i00033b6iil8qiu5c%2522%253A%257B%2522tabs%2522%253A%255B%257B%2522id%2522%253A%2522clsqxqd6i00023b6ium36wr5i%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522type%2522%253A%2522FILE%2522%252C%2522initialSelections%2522%253A%255B%257B%2522startLineNumber%2522%253A22%252C%2522startColumn%2522%253A34%252C%2522endLineNumber%2522%253A22%252C%2522endColumn%2522%253A34%257D%255D%252C%2522filepath%2522%253A%2522%252Fsrc%252FApp.js%2522%252C%2522state%2522%253A%2522IDLE%2522%257D%255D%252C%2522id%2522%253A%2522clsqxqd6i00033b6iil8qiu5c%2522%252C%2522activeTabId%2522%253A%2522clsqxqd6i00023b6ium36wr5i%2522%257D%252C%2522clsqxqd6i00073b6ie9mwiyfc%2522%253A%257B%2522tabs%2522%253A%255B%257B%2522id%2522%253A%2522clsqxqd6i00063b6i7w6hursl%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522type%2522%253A%2522TASK_PORT%2522%252C%2522taskId%2522%253A%2522start%2522%252C%2522port%2522%253A3000%252C%2522path%2522%253A%2522%252F%2522%257D%255D%252C%2522id%2522%253A%2522clsqxqd6i00073b6ie9mwiyfc%2522%252C%2522activeTabId%2522%253A%2522clsqxqd6i00063b6i7w6hursl%2522%257D%252C%2522clsqxqd6i00053b6ivknllgky%2522%253A%257B%2522tabs%2522%253A%255B%257B%2522id%2522%253A%2522clsqxqd6i00043b6ijrxc569y%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522type%2522%253A%2522TASK_LOG%2522%252C%2522taskId%2522%253A%2522start%2522%257D%255D%252C%2522id%2522%253A%2522clsqxqd6i00053b6ivknllgky%2522%252C%2522activeTabId%2522%253A%2522clsqxqd6i00043b6ijrxc569y%2522%257D%257D%252C%2522showDevtools%2522%253Atrue%252C%2522showShells%2522%253Atrue%252C%2522showSidebar%2522%253Atrue%252C%2522sidebarPanelSize%2522%253A15%257D).

> Looks some of the jointjs core components are not working while react state is updated.

What do you mean?

**Reply by rajeshK88** [reply]
> About double mesh example  [example](https://codesandbox.io/p/devbox/baalakay-node-react-forked-2hhcy5?file=%2Fsrc%2FApp.js%3A22%2C34&layout=%257B%2522sidebarPanel%2522%253A%2522EXPLORER%2522%252C%2522rootPanelGroup%2522%253A%257B%2522direction%2522%253A%2522horizontal%2522%252C%2522contentType%2522%253A%2522UNKNOWN%2522%252C%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522id%2522%253A%2522ROOT_LAYOUT%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522UNKNOWN%2522%252C%2522direction%2522%253A%2522vertical%2522%252C%2522id%2522%253A%2522clsqxqd6i00083b6itdcrmxdz%2522%252C%2522sizes%2522%253A%255B70%252C30%255D%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522EDITOR%2522%252C%2522direction%2522%253A%2522horizontal%2522%252C%2522id%2522%253A%2522EDITOR%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522EDITOR%2522%252C%2522id%2522%253A%2522clsqxqd6i00033b6iil8qiu5c%2522%257D%255D%257D%252C%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522SHELLS%2522%252C%2522direction%2522%253A%2522horizontal%2522%252C%2522id%2522%253A%2522SHELLS%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522SHELLS%2522%252C%2522id%2522%253A%2522clsqxqd6i00053b6ivknllgky%2522%257D%255D%252C%2522sizes%2522%253A%255B100%255D%257D%255D%257D%252C%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522DEVTOOLS%2522%252C%2522direction%2522%253A%2522vertical%2522%252C%2522id%2522%253A%2522DEVTOOLS%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522DEVTOOLS%2522%252C%2522id%2522%253A%2522clsqxqd6i00073b6ie9mwiyfc%2522%257D%255D%252C%2522sizes%2522%253A%255B100%255D%257D%255D%252C%2522sizes%2522%253A%255B50%252C50%255D%257D%252C%2522tabbedPanels%2522%253A%257B%2522clsqxqd6i00033b6iil8qiu5c%2522%253A%257B%2522tabs%2522%253A%255B%257B%2522id%2522%253A%2522clsqxqd6i00023b6ium36wr5i%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522type%2522%253A%2522FILE%2522%252C%2522initialSelections%2522%253A%255B%257B%2522startLineNumber%2522%253A22%252C%2522startColumn%2522%253A34%252C%2522endLineNumber%2522%253A22%252C%2522endColumn%2522%253A34%257D%255D%252C%2522filepath%2522%253A%2522%252Fsrc%252FApp.js%2522%252C%2522state%2522%253A%2522IDLE%2522%257D%255D%252C%2522id%2522%253A%2522clsqxqd6i00033b6iil8qiu5c%2522%252C%2522activeTabId%2522%253A%2522clsqxqd6i00023b6ium36wr5i%2522%257D%252C%2522clsqxqd6i00073b6ie9mwiyfc%2522%253A%257B%2522tabs%2522%253A%255B%257B%2522id%2522%253A%2522clsqxqd6i00063b6i7w6hursl%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522type%2522%253A%2522TASK_PORT%2522%252C%2522taskId%2522%253A%2522start%2522%252C%2522port%2522%253A3000%252C%2522path%2522%253A%2522%252F%2522%257D%255D%252C%2522id%2522%253A%2522clsqxqd6i00073b6ie9mwiyfc%2522%252C%2522activeTabId%2522%253A%2522clsqxqd6i00063b6i7w6hursl%2522%257D%252C%2522clsqxqd6i00053b6ivknllgky%2522%253A%257B%2522tabs%2522%253A%255B%257B%2522id%2522%253A%2522clsqxqd6i00043b6ijrxc569y%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522type%2522%253A%2522TASK_LOG%2522%252C%2522taskId%2522%253A%2522start%2522%257D%255D%252C%2522id%2522%253A%2522clsqxqd6i00053b6ivknllgky%2522%252C%2522activeTabId%2522%253A%2522clsqxqd6i00043b6ijrxc569y%2522%257D%257D%252C%2522showDevtools%2522%253Atrue%252C%2522showShells%2522%253Atrue%252C%2522showSidebar%2522%253Atrue%252C%2522sidebarPanelSize%2522%253A15%257D). 

looks like you imported joint-plus.css . So do we need to import this every time. Is there any css module available in @joint/core to import.
 
 > Looks some of the jointjs core components are not working while react state is updated.
    
When the reactjs state is changed and paper refreshes again then we loose the graph component update. Will create codesandbox and share it here.

**Reply by kumilingus** [reply]
The `joint-plus.css` was a leftover. I updated the example. There is no CSS for `@joint/core` required.
